src/s3fs.cpp

```diff
@@ -393,78 +402,24 @@ static int get_object_attribute(const char* path, struct stat* pstbuf, headers_t
-        // Thus, set the mode of 0000 here in the meta header so
-        // that s3fs can print a permission error when the file
-        // is actually accessed.
-        // It is better not to set meta header other than mode,
-        // so do not do it.
-        //
-        (*pheader)["x-amz-meta-mode"] = str(0);
-
-    }else if(0 != result){
-        if(overcheck){
-            // when support_compat_dir is disabled, strpath maybe have "_$folder$".
-            if('/' != strpath[strpath.length() - 1] && std::string::npos == strpath.find("_$folder$", 0)){
-                // now path is "object", do check "object/" for over checking
-                strpath    += "/";
-                result      = s3fscurl.HeadRequest(strpath.c_str(), (*pheader));
-                s3fscurl.DestroyCurlHandle();
-            }
-            if(support_compat_dir && 0 != result){
-                // now path is "object/", do check "object_$folder$" for over checking
-                strpath.erase(strpath.length() - 1);
-                strpath    += "_$folder$";
-                result      = s3fscurl.HeadRequest(strpath.c_str(), (*pheader));
-                s3fscurl.DestroyCurlHandle();
-
-              if(0 != result){
-                  // cut "_$folder$" for over checking "no dir object" after here
-                  if(std::string::npos != (Pos = strpath.find("_$folder$", 0))){
-                      strpath.erase(Pos);
-                  }
-              }
-            }
-        }
-        if(support_compat_dir && 0 != result && std::string::npos == strpath.find("_$folder$", 0)){
-            // now path is "object" or "object/", do check "no dir object" which is not object but has only children.
-            if('/' == strpath[strpath.length() - 1]){
-                strpath.erase(strpath.length() - 1);
-            }
-            if(-ENOTEMPTY == directory_empty(strpath.c_str())){
-                // found "no dir object".
-                strpath  += "/";
-                *pisforce = true;
-                result    = 0;
-            }
-        }
+    s3obj_list_t headlist;
+    std::string parent;
+    std::string child;
+    size_t pos = strpath.rfind('/');
+    if(pos == std::string::npos){
+        parent = "/";
+        child = strpath;
     }else{
-        if(support_compat_dir && '/' != strpath[strpath.length() - 1] && std::string::npos == strpath.find("_$folder$", 0) && is_need_check_obj_detail(*pheader)){
-            // check a case of that "object" does not have attribute and "object" is possible to be directory.
-            if(-ENOTEMPTY == directory_empty(strpath.c_str())){
-                // found "no dir object".
-                strpath  += "/";
-                *pisforce = true;
-                result    = 0;
-            }
-        }
+        parent = strpath.substr(0, pos + 1);
+        child = strpath.substr(pos + 1);
+    }
+    headlist.push_back(child);
+    headlist.push_back(child + "/");
```

Comment:
I request path, path/, and path_$folder$ in parallel, but it seems that the priority is unclear if one or more of them exist.
(We need to consider whether that situation exists, because we are parallelizing from sequential.)
At present, it seems that the content that received the response (normal) at the very end is registered as stat.

Reply:
I will add a test for this.  If there exists `path` and `path/` the current behavior is to return `path`.